modules/core/src/main/java/org/apache/synapse/message/processor/impl/ScheduledMessageProcessor.java

```diff
@@ -63,342 +54,273 @@
-            if (o != null) {
-                isActivated.set(Boolean.valueOf(o.toString()));
+            param = parameters.get(MessageProcessorConstants.IS_ACTIVATED);
+            if (param != null) {
+                setActivated(Boolean.valueOf(param.toString()));
             }
-            o = parameters.get(ForwardingProcessorConstants.NON_RETRY_STATUS_CODES);
-            if (o != null) {
-                // we take it out of param set and send it because we need split the array.
-                nonRetryStatusCodes = o.toString().split(",");
+            param = parameters.get(ForwardingProcessorConstants.NON_RETRY_STATUS_CODES);
+            if (param != null) {
+                // we take it out of param set and send it because we need split
+                // the array.
+                nonRetryStatusCodes = param.toString().split(",");
             }
         }
     }
 
-    private JobBuilder getJobBuilder() {
-        // This is just to set the default one
-        JobBuilder jobBuilder;
-
-        if (this instanceof SamplingProcessor) {
-            jobBuilder = JobBuilder.newJob(SamplingService.class);
-        }
-        else {
-            jobBuilder = JobBuilder.newJob(ForwardingService.class);
-        }
-
-        jobBuilder.withIdentity(name + "-job", MessageProcessorConstants.SCHEDULED_MESSAGE_PROCESSOR_GROUP);
-
-        return jobBuilder;
-    }
-
-    protected JobDataMap getJobDataMap() {
-        return new JobDataMap();
-    }
-
     public boolean stop() {
-        try {
-            if (scheduler != null) {
-                // There could be servers that are disabled at startup time. therefore not started but initiated.
-                if (scheduler.isStarted()) {
-                    // This is to immediately stop the scheduler to avoid firing new services
-                    scheduler.standby();
-
-                    if (logger.isDebugEnabled()) {
-                        logger.debug("ShuttingDown Message Processor Scheduler : " + scheduler.getMetaData());
-                    }
-
-                    try {
-                        scheduler.interrupt(new JobKey(name + "-job", MessageProcessorConstants.SCHEDULED_MESSAGE_PROCESSOR_GROUP));
-                    } catch (UnableToInterruptJobException e) {
-                        logger.info("Unable to interrupt job [" + name + "-job]");
-                    }
+        /*
+         * There could be servers that are disabled at startup time.
+         * therefore not started but initiated.
+         */
+        if (taskManager != null && taskManager.isInitialized()) {
+            for (int i = 0; i < memberCount; i++) {
+                /*
+                 * This is to immediately stop the scheduler to avoid firing new
+                 * services
+                 */
+                taskManager.pause(TASK_PREFIX + name + i);
+                if (logger.isDebugEnabled()) {
+                    logger.debug("ShuttingDown Message Processor Scheduler : " +
```

Comment:
It might be better to change the debug log message to "Shutting down message processor scheduler: "